### PR TITLE
Facet distribution correct case

### LINKED ISSUE #786: Keys are lowercased in facet distribution
When asking for a facet distribution, the keys are returned lowercased. They should be returned with the case specified by the user.

### PR DESCRIPTION
**Pull request by MarinPostma**
fix all facets being lowercased in facet distribution
close #786

### CODE REVIEW

File: meilisearch-core/src/facets.rs [hunk 1]
```diff
@@ -185,8 +185,10 @@ pub fn facet_map_from_docids(
     index: &crate::Index,
     document_ids: &[DocumentId],
     attributes_for_facetting: &[FieldId],
-) -> MResult<HashMap<FacetKey, Vec<DocumentId>>> {
-    let mut facet_map = HashMap::new();
+) -> MResult<HashMap<FacetKey, (String, Vec<DocumentId>)>> {
+    // A hashmap that ascociate a facet key to a pair containing the original facet attribute
+    // string with it's case preserved, and a list of document ids for that facet attribute.
+    let mut facet_map: HashMap<FacetKey, (String, Vec<DocumentId>)> = HashMap::new();
```
Comment (MarinPostma) [inline meilisearch-core/src/facets.rs:191]: the type is hinted for ease of understanding only.

File: meilisearch-core/src/query_builder.rs [hunk 2]
```diff
@@ -100,14 +100,14 @@ impl<'c, 'f, 'd, 'i> QueryBuilder<'c, 'f, 'd, 'i> {
                                 let docids = self.index.facets.facet_document_ids(reader, &key)?.unwrap_or_default();
                                 ors.push(docids);
                             }
-                            let sets: Vec<_> = ors.iter().map(Cow::deref).collect();
+                            let sets: Vec<_> = ors.iter().map(|i| &i.1).map(Cow::deref).collect();
```
Comment (LegendreM) [inline meilisearch-core/src/query_builder.rs:103]: `|i| &i.1` -> `|(_, i)| &i`

File: meilisearch-core/src/query_builder.rs [hunk 3]
```diff
@@ -100,14 +100,14 @@ impl<'c, 'f, 'd, 'i> QueryBuilder<'c, 'f, 'd, 'i> {
...
                             let or_result = sdset::multi::OpBuilder::from_vec(sets).union().into_set_buf();
                             ands.push(Cow::Owned(or_result));
                             ors.clear();
                         }
                         Either::Right(key) =>{
                             match self.index.facets.facet_document_ids(reader, &key)? {
-                                Some(docids) => ands.push(docids),
+                                Some(docids) => ands.push(docids.1),
```
Comment (LegendreM) [inline meilisearch-core/src/query_builder.rs:110]: ` Some(docids) => ands.push(docids.1),` -> ` Some((_, docids)) => ands.push(docids),`

File: meilisearch-core/src/query_builder.rs [hunk 4]
```diff
@@ -97,16 +97,14 @@ impl<'c, 'f, 'd, 'i> QueryBuilder<'c, 'f, 'd, 'i> {
...
-                                .into_set_buf();
+                            let sets: Vec<_> = ors.iter().map(|(_, i)| i).map(Cow::deref).collect();
+                            let or_result = sdset::multi::OpBuilder::from_vec(sets).union().into_set_buf();
                             ands.push(Cow::Owned(or_result));
                             ors.clear();
                         }
                         Either::Right(key) => {
                             match self.index.facets.facet_document_ids(reader, &key)? {
-                                Some(docids) => ands.push(docids),
+                                Some((_name, docids)) => ands.push(docids),
```
Comment (LegendreM) [inline meilisearch-core/src/query_builder.rs:107]: `_` or `_names` ?

**Comment by MarinPostma** [comment]
bors merge

**Pull request merged**